src/hotspot/share/gc/g1/g1BlockOffsetTable.hpp

```diff
@@ -33,26 +33,27 @@
 #include "utilities/globalDefinitions.hpp"
 
 // Forward declarations
-class G1BlockOffsetTable;
 class HeapRegion;
 
 // This implementation of "G1BlockOffsetTable" divides the covered region
 // into "N"-word subregions (where "N" = 2^"LogN".  An array with an entry
 // for each such subregion indicates how far back one must go to find the
 // start of the chunk that includes the first word of the subregion.
 //
-// Each G1BlockOffsetTablePart is owned by a HeapRegion.
+// Each G1BlockOffsetTable is owned by a HeapRegion.
 
 class G1BlockOffsetTable: public CHeapObj<mtGC> {
-  friend class G1BlockOffsetTablePart;
   friend class VMStructs;
 
 private:
   // The reserved region covered by the table.
-  MemRegion _reserved;
+  static MemRegion _reserved;
 
   // Biased array-start of BOT array for fast BOT entry translation
-  volatile uint8_t* _offset_base;
+  static volatile uint8_t* _offset_base;
+
+  // The region that owns this BOT.
+  HeapRegion* _hr;
```

Comment:
I marked the fields `_offset_base` and `_reserved` as `non-static`, removed the field `_hr` and shared a simple/unified BOT for all the heap regions. Since few methods need the heap region, I used the heap region as method parameter in such methods.